src/app/shared/util/metrics.util.ts

```diff
@@ -0,0 +1,57 @@
+import { PodcastModel, EpisodeModel, FilterModel, MetricsType, PodcastMetricsModel, EpisodeMetricsModel } from '../../ngrx/model';
+
```

Comment:
I separated these filtering and utility functions from the selectors for better organization.